packages/ag-charts-community/src/chart/dom/domManager.ts

```diff
@@ -336,16 +349,8 @@ export class DOMManager extends BaseManager<Events['type'], Events> {
     }
 
     isEventOverElement(event: Event | MouseEvent | TouchEvent) {
-        let element = event.target;
-
-        if (element == null) return false;
-
-        while (element !== this.element) {
-            element = (element as HTMLElement).parentElement;
-            if (element == null) return false;
-        }
-
-        return true;
+        const element = event.target as HTMLElement | null;
+        return element != null && this.element.contains(element);
```

Comment:
I'd prefer to keep the previous implementation, unless we have strong guarantees that `Element.contains()` is going to be faster then traversing parents, which is typically going to be faster than a top-down tree traversal?

Reply:
Element.contains does the same thing we did - it just checks the parents. If you were interested - https://github.com/WebKit/WebKit/blob/1b86aba5013c21c198a3182424ec5c10b585aeb6/Source/WebCore/dom/Node.cpp#L1198-L1201